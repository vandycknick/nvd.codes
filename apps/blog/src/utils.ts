import {
  AnyRouter,
  CreateContextFn,
  CreateContextFnOptions,
  requestHandler,
} from "@trpc/server"
import { Middleware } from "koa"
import { IncomingMessage, ServerResponse } from "http"

export const getEnvVar = (key: string, defaultz?: string): string => {
  const result = process.env[key]

  if (result == undefined && defaultz != undefined) {
    return defaultz
  } else if (result != undefined) {
    return result
  }

  throw new Error(`Required environment variable '${key}' is not defined!`)
}

export const getEnvVarAsInt = (key: string, defaultz: number): number => {
  const result = getEnvVar(key, "")

  if (result == "") {
    return defaultz
  }

  return parseInt(result, 10)
}

export const getOptionalEnvVar = (key: string, defaultz: string) => {
  const result = process.env[key]

  if (result != undefined) {
    return result
  }

  return defaultz
}

export type CreateKoaContextOptions = CreateContextFnOptions<
  IncomingMessage,
  ServerResponse
>

export type CreateKoaContextFn<TRouter extends AnyRouter> = CreateContextFn<
  TRouter,
  IncomingMessage,
  ServerResponse
>

export const createKoaMiddleware =
  <TRouter extends AnyRouter>({
    router,
    createContext,
  }: {
    router: TRouter
    createContext: CreateKoaContextFn<TRouter>
  }): Middleware =>
  async (context) => {
    const endpoint = context.request.path.substr(1) ?? ""

    await requestHandler<
      TRouter,
      CreateKoaContextFn<TRouter>,
      IncomingMessage,
      ServerResponse
    >({
      createContext,
      req: context.req,
      res: context.res,
      path: endpoint,
      router,
    })
  }

export interface MemoizedFunction {
  cache: Map<unknown, unknown>
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const memoize = <T extends (...args: any[]) => any>(
  func: T,
): T & MemoizedFunction => {
  if (typeof func !== "function") {
    throw new TypeError("Expected a function!")
  }

  const memoized = Object.assign(
    (...args: unknown[]) => {
      const key = args[0]
      const cache = memoized.cache

      if (cache.has(key)) {
        return cache.get(key)
      }
      const result = func.apply(this, args)
      memoized.cache = cache.set(key, result) || cache
      return result
    },
    {
      cache: new Map<unknown, unknown>(),
    },
  )

  return memoized as T & MemoizedFunction
}
