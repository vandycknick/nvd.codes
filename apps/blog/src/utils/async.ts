/* eslint-disable @typescript-eslint/no-explicit-any */
class HttpError extends Error {
  constructor(public statusCode: number, public msg: string) {
    super(msg)
  }
}

const fetchJSON = async <T>(url: string): Promise<T> => {
  const response = await fetch(url)
  if (!response.ok) {
    throw new HttpError(response.status, response.statusText)
  }
  const data = await response.json()
  return data
}

type Resolver<T> = (value: T | PromiseLike<T>) => void
type Rejector<T> = (reason?: T) => void

class Deferred<T> implements Promise<T> {
  private _resolveSelf: Resolver<T> = Promise.resolve
  private _rejectSelf: Rejector<T> = Promise.reject
  private _promise: Promise<T>

  public constructor() {
    this._promise = new Promise<T>((resolve, reject): void => {
      this._resolveSelf = resolve
      this._rejectSelf = reject
    })
  }

  public then<TResult1 = T, TResult2 = never>(
    onfulfilled?:
      | ((value: T) => TResult1 | PromiseLike<TResult1>)
      | undefined
      | null,
    onrejected?:
      | ((reason: any) => TResult2 | PromiseLike<TResult2>)
      | undefined
      | null,
  ): Promise<TResult1 | TResult2> {
    return this._promise.then(onfulfilled, onrejected)
  }

  public catch<TResult = never>(
    onrejected?:
      | ((reason: any) => TResult | PromiseLike<TResult>)
      | undefined
      | null,
  ): Promise<T | TResult> {
    return this._promise.catch(onrejected)
  }

  public finally(onfinally?: (() => void) | null | undefined): Promise<T> {
    return this._promise.finally(onfinally)
  }

  public resolve(val: T): void {
    this._resolveSelf(val)
  }

  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
  public reject(reason: any): void {
    this._rejectSelf(reason)
  }

  public [Symbol.toStringTag]: "Promise" = "Promise"
}

export { fetchJSON, Deferred }
