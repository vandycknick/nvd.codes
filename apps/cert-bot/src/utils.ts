import { None, Option, Some, Ok, Err } from "@nvd.codes/monad"
import { asn1, pkcs12, pki, util } from "node-forge"
import { promisify } from "util"
import { KeyVaultCertificateResponse } from "./models/keyVault"

const generateKeyPair = promisify(pki.rsa.generateKeyPair)

export const getEnvVar = (key: string): Option<string> => {
  const result = process.env[key]
  return result === undefined ? None() : Some(result)
}

export const getOptionalEnvVar = (
  key: string,
  defaultz: string,
): Option<string> => {
  const result = getEnvVar(key)
  return result.isNone() ? Some(defaultz) : result
}

export const createPrivateKey = async (size = 2048) => {
  const keyPair = await generateKeyPair({ bits: size })
  const pemKey = pki.privateKeyToPem(keyPair.privateKey)
  return pemKey
}

export const parseKeyVaultCertificate = (cert: KeyVaultCertificateResponse) => {
  const pem = `-----BEGIN CERTIFICATE-----\n${cert.cer}\n-----END CERTIFICATE-----`
  return pki.certificateFromPem(pem)
}

export const convertToPkcs12Certificate = (cert: string, key: string) => {
  const asn = pkcs12.toPkcs12Asn1(
    pki.privateKeyFromPem(key),
    pki.certificateFromPem(cert),
    null,
  )

  const der = asn1.toDer(asn).getBytes()
  return util.encode64(der)
}

export const privateKeyToPkcs8 = (key: string) => {
  // convert a Forge private key to an ASN.1 RSAPrivateKey
  const forgePrivateKey = pki.privateKeyFromPem(key.toString())

  // convert a Forge private key to an ASN.1 RSAPrivateKey
  const rsaPrivateKey = pki.privateKeyToAsn1(forgePrivateKey)

  // wrap an RSAPrivateKey ASN.1 object in a PKCS#8 ASN.1 PrivateKeyInfo
  const privateKeyInfo = pki.wrapRsaPrivateKey(rsaPrivateKey)

  // convert a PKCS#8 ASN.1 PrivateKeyInfo to PEM
  const pem = pki.privateKeyInfoToPem(privateKeyInfo)
  return pem
}

export const delay = (time: number): Promise<void> =>
  new Promise((res) => {
    setTimeout(() => res(), time)
  })

export const parseAzureResourceId = (resourceId: string) => {
  const resourceIdPattern = /subscriptions\/(.+)\/resourceGroups\/(.+)\/providers\/(.+?)\/(.+?)\/(.+)/gi

  const matches = resourceIdPattern.exec(resourceId)

  if (matches == undefined || matches.length === 0) {
    return Err(`Failed parsing for ${resourceId}, invalid resourceId format!`)
  }

  const chunks = matches[5]?.split("/")
  const resourceName = chunks.pop()
  const resourceType = chunks.pop() ?? matches[4]

  return Ok({
    subscriptionId: matches[1],
    resourceGroupName: matches[2],
    provider: matches[3],
    resourceType,
    resourceName,
    uri: `${matches[4]}/${matches[5]}`,
  })
}

interface MemoizedFunction {
  cache: Map<unknown, unknown>
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const memoize = <T extends (...args: any[]) => any>(
  func: T,
): T & MemoizedFunction => {
  if (typeof func !== "function") {
    throw new TypeError("Expected a function!")
  }

  const memoized = Object.assign(
    (...args: unknown[]) => {
      const key = args[0]
      const cache = memoized.cache

      if (cache.has(key)) {
        return cache.get(key)
      }
      const result = func.apply(this, args)
      memoized.cache = cache.set(key, result) || cache
      return result
    },
    {
      cache: new Map<unknown, unknown>(),
    },
  )

  return memoized as T & MemoizedFunction
}
