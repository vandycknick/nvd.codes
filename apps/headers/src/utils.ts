type Functor = (...args: unknown[]) => unknown

// Overloads stolen from: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d6194f0cc57d1bf1c86d00cf9a1f4c6207f2bf12/types/ramda/index.d.ts#L1420
export function pipe<T1>(fn0: () => T1): () => T1
export function pipe<V0, T1>(fn0: (x0: V0) => T1): (x0: V0) => T1

export function pipe<T1, T2>(fn0: () => T1, fn1: (x: T1) => T2): () => T2
export function pipe<V0, T1, T2>(
  fn0: (x0: V0) => T1,
  fn1: (x: T1) => T2,
): (x0: V0) => T2

export function pipe<T1, T2, T3>(
  fn0: () => T1,
  fn1: (x: T1) => T2,
  fn2: (x: T2) => T3,
): () => T3
export function pipe<V0, T1, T2, T3>(
  fn0: (x: V0) => T1,
  fn1: (x: T1) => T2,
  fn2: (x: T2) => T3,
): (x: V0) => T3

export function pipe(...fns: Functor[]) {
  return (v: unknown) => fns.reduce((v, f) => f(v), v)
}
