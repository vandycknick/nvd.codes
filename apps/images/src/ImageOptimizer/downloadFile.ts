import { BlobServiceClient, BlockBlobClient } from "@azure/storage-blob"
import { getConfig } from "../config"

interface File {
  contentType?: string
  contents: Buffer
}

const readStream = (stream: NodeJS.ReadableStream): Promise<Buffer> => {
  return new Promise((resolve, reject) => {
    const data: Uint8Array[] = []

    stream.on("data", (chunk) => {
      data.push(chunk)
    })
    stream.on("end", () => {
      resolve(Buffer.concat(data))
    })
    stream.on("error", (error) => reject(error))
  })
}

const getBlockBlobClient = (
  container: string,
  filePath: string,
): BlockBlobClient => {
  const config = getConfig()
  const blobService = BlobServiceClient.fromConnectionString(
    config.AZURE_STORAGE_CONNECTION_STRING,
  )
  const containerClient = blobService.getContainerClient(container)
  const blockBlobClient = containerClient.getBlockBlobClient(filePath)
  return blockBlobClient
}

export const downloadFile = async (
  container: string,
  filePath: string,
): Promise<File | undefined> => {
  const blockBlobClient = getBlockBlobClient(container, filePath)
  const file = await blockBlobClient.download(0)

  if (file == undefined || file.readableStreamBody == undefined) {
    return undefined
  }

  const contents = await readStream(file.readableStreamBody)

  return {
    contentType: file.contentType,
    contents,
  }
}
