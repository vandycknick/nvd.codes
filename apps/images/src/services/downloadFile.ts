import { Region, SimpleAuthenticationDetailsProvider } from "oci-common"
import { ObjectStorageClient } from "oci-objectstorage"

import { getConfig } from "../config"

interface File {
  contentType?: string
  contents: Buffer
}

const readStream = (stream: NodeJS.ReadableStream): Promise<Buffer> => {
  return new Promise((resolve, reject) => {
    const data: Uint8Array[] = []

    stream.on("data", (chunk) => {
      data.push(chunk)
    })
    stream.on("end", () => {
      resolve(Buffer.concat(data))
    })
    stream.on("error", (error) => reject(error))
  })
}

const config = getConfig()

const provider = new SimpleAuthenticationDetailsProvider(
  config.ociTenancy,
  config.ociUser,
  config.ociFingerprint,
  config.ociPrivateKey,
  null,
  Region.fromRegionId(config.ociRegion),
)
const client = new ObjectStorageClient({
  authenticationDetailsProvider: provider,
})

export const downloadFile = async (
  filePath: string,
): Promise<File | undefined> => {
  const config = getConfig()
  const response = await client.getObject({
    bucketName: config.bucketName,
    namespaceName: config.bucketNamespace,
    objectName: filePath,
  })

  if (response.value == null) return undefined

  const stream = response.value as NodeJS.ReadableStream

  const contents = await readStream(stream)

  return {
    contentType: response.contentType,
    contents,
  }
}
