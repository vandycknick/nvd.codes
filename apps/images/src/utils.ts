import { BindingDefinition, Context, Logger } from "@azure/functions"

export const createMockContext = (
  bindingDefinitions: unknown[],
): jest.Mocked<Context> => {
  const log = (jest.fn() as unknown) as jest.MockInstance<void, []> & Logger

  log.verbose = jest.fn()
  log.info = jest.fn()
  log.warn = jest.fn()
  log.error = jest.fn()
  return {
    invocationId: "123",
    executionContext: {
      invocationId: "123",
      functionName: "ImageOptimizer",
      functionDirectory: "ImageOptimizer",
    },
    bindings: {},
    bindingData: {},
    traceContext: {
      traceparent: undefined,
      tracestate: undefined,
      attributes: {},
    },
    bindingDefinitions: bindingDefinitions as BindingDefinition[],
    log,
    done: jest.fn(),
  }
}

const sufixes = ["Bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"]
export const byteSize = (bytes: number): string => {
  const i = Math.floor(Math.log(bytes) / Math.log(1024))
  return (
    (!bytes && "0 Bytes") ||
    (bytes / Math.pow(1024, i)).toFixed(2) + " " + sufixes[i]
  )
}
