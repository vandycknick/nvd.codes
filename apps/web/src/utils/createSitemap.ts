import { SitemapStream, streamToPromise } from "sitemap"
import { Readable } from "stream"
import { join } from "path"
import { promises } from "fs"

import { listAllPosts } from "../services/posts"

const { writeFile } = promises
const DIST = ".dist"

const links = ["/", "/blog", "/about"]
const stream = new SitemapStream({ hostname: "https://nvd.codes" })

const createSitemap = async () => {
  const posts = await listAllPosts()
  const slugs = posts.map((post) => `/post/${post.slug}`)
  const all = ([] as string[]).concat(links, slugs)
  const sitemap = await streamToPromise(Readable.from(all).pipe(stream))
  return sitemap.toString()
}

const main = async () => {
  const sitemap = await createSitemap()
  const filepath = join(process.cwd(), DIST, "sitemap.xml")

  await writeFile(filepath, sitemap)
}

main()
