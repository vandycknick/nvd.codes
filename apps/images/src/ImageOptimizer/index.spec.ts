import imageOptimizer from "./index"
import functionJson from "./function.json"
import { createMockContext } from "../utils"
import { downloadFile } from "./downloadFile"

jest.mock("./downloadFile")
jest.mock("../config.ts", () => ({
  getConfig: jest.fn(() => ({
    AZURE_STORAGE_CONNECTION_STRING: "one",
    IMAGES_CONTAINER: "conainer",
  })),
}))

const createImageOptimizerMockContext = () => {
  return createMockContext(functionJson.bindings)
}

describe("imageOptimizer", () => {
  const downloadFileMock = downloadFile as jest.MockedFunction<
    typeof downloadFile
  >

  beforeEach(() => {
    const buffer = Buffer.from(
      "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP87CPxCQAF3AJL+UURowAAAABJRU5ErkJggg==",
      "base64",
    )
    downloadFileMock.mockReset()
    downloadFileMock.mockResolvedValue({
      contentType: "image/png",
      contents: buffer,
    })
  })

  it("should return a 404 when req is undefined", async () => {
    // Given
    const context = createImageOptimizerMockContext()

    // When
    const response = await imageOptimizer(context)

    // Then
    expect(response).toEqual({
      status: 404,
      body: "Not Found",
    })
  })

  it.each`
    method      | status | body
    ${"POST"}   | ${405} | ${"Method not allowed"}
    ${"DELETE"} | ${405} | ${"Method not allowed"}
    ${"PATCH"}  | ${405} | ${"Method not allowed"}
    ${"PUT"}    | ${405} | ${"Method not allowed"}
    ${"TRACE"}  | ${405} | ${"Method not allowed"}
  `(
    "should return a $status when req is a $method method",
    async ({ method, status, body }) => {
      // Given
      const context = createImageOptimizerMockContext()
      context.req = {
        method,
        url: "/_next/images/one.jpg",
        headers: {},
        query: {},
        params: {},
      }

      // When
      const response = await imageOptimizer(context)

      // Then
      expect(response).toEqual({
        status,
        body,
      })
    },
  )

  it.each`
    imagePath                | status | body
    ${undefined}             | ${404} | ${"Not Found"}
    ${""}                    | ${404} | ${"Not Found"}
    ${"subfolder"}           | ${404} | ${"Not Found"}
    ${"subfolder/subfolder"} | ${404} | ${"Not Found"}
  `(
    "should return a $status when imagePath parameter is '$imagePath'",
    async ({ imagePath, status, body }) => {
      // Given
      const context = createImageOptimizerMockContext()
      context.req = {
        method: "GET",
        url: `/_next/images/${imagePath ?? ""}`,
        headers: {},
        query: {},
        params: { imagePath },
      }

      // When
      const response = await imageOptimizer(context)

      // Then
      expect(response).toEqual({
        status,
        body,
      })
    },
  )

  it("should return a 400 when no width query string is provided", async () => {
    // Given
    const context = createImageOptimizerMockContext()
    const imagePath = "image.jpeg"
    context.req = {
      method: "GET",
      url: `/_next/images/${imagePath ?? ""}`,
      headers: {},
      query: {},
      params: { imagePath },
    }

    // When
    const response = await imageOptimizer(context)

    // Then
    expect(response).toEqual({
      status: 400,
      body: "'w' parameter (width) is required",
    })
  })

  it.each`
    width         | status | body
    ${"OO10"}     | ${400} | ${"'w' parameter (width) must be a number greater than 0"}
    ${"string"}   | ${400} | ${"'w' parameter (width) must be a number greater than 0"}
    ${"Infinity"} | ${400} | ${"'w' parameter (width) must be a number greater than 0"}
    ${"0x10"}     | ${400} | ${"'w' parameter (width) must be a number greater than 0"}
    ${"-32879"}   | ${400} | ${"'w' parameter (width) of -32879 is not allowed"}
    ${"266"}      | ${400} | ${"'w' parameter (width) of 266 is not allowed"}
    ${"644"}      | ${400} | ${"'w' parameter (width) of 644 is not allowed"}
    ${"4000"}     | ${400} | ${"'w' parameter (width) of 4000 is not allowed"}
  `(
    "should return a $status when given $width as width query string",
    async ({ width, status, body }) => {
      // Given
      const context = createImageOptimizerMockContext()
      const imagePath = "image.jpeg"
      context.req = {
        method: "GET",
        url: `/_next/images/${imagePath ?? ""}?w=${width}`,
        headers: {},
        query: { w: width },
        params: { imagePath },
      }

      // When
      const response = await imageOptimizer(context)

      // Then
      expect(response).toEqual({
        status,
        body,
      })
    },
  )

  it("should return a 404 when downloadFile returns undefined", async () => {
    // Given
    downloadFileMock.mockReset()
    downloadFileMock.mockResolvedValue(undefined)
    const context = createImageOptimizerMockContext()
    const imagePath = "image.png"
    const width = "256"
    context.req = {
      method: "GET",
      url: `/_next/images/${imagePath ?? ""}?w=${width}`,
      headers: {},
      query: { w: width },
      params: { imagePath },
    }

    // When
    const response = await imageOptimizer(context)

    // Then
    expect(response).toEqual({
      status: 404,
      body: "Not Found",
    })
  })

  it("should return a 200 image jpeg result when no content-type is given", async () => {
    // Given
    const buffer = Buffer.from(
      "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP87CPxCQAF3AJL+UURowAAAABJRU5ErkJggg==",
      "base64",
    )
    downloadFileMock.mockReset()
    downloadFileMock.mockResolvedValue({
      contents: buffer,
    })
    const context = createImageOptimizerMockContext()
    const imagePath = "image.jpeg"
    const width = "256"
    context.req = {
      method: "GET",
      url: `/_next/images/${imagePath ?? ""}?w=${width}`,
      headers: {},
      query: { w: width },
      params: { imagePath },
    }

    // When
    const response = await imageOptimizer(context)

    // Then
    expect(response).toMatchSnapshot()
  })

  it.each`
    type
    ${"jpeg"}
    ${"png"}
    ${"unknown"}
    ${"webp"}
  `(
    "should return a 200 image $type result when no content-type is given but the filetype is $type",
    async ({ type }) => {
      // Given
      const buffer = Buffer.from(
        "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP87CPxCQAF3AJL+UURowAAAABJRU5ErkJggg==",
        "base64",
      )
      downloadFileMock.mockReset()
      downloadFileMock.mockResolvedValue({
        contentType: `image/${type}`,
        contents: buffer,
      })
      const context = createImageOptimizerMockContext()
      const imagePath = "image.jpeg"
      const width = "16"
      context.req = {
        method: "GET",
        url: `/_next/images/${imagePath ?? ""}?w=${width}`,
        headers: {},
        query: { w: width },
        params: { imagePath },
      }

      // When
      const response = await imageOptimizer(context)

      // Then
      expect(response).toMatchSnapshot()
    },
  )

  it("should return a 200 image webp result when the request contains an accept header", async () => {
    // Given
    const context = createImageOptimizerMockContext()
    const imagePath = "image.jpeg"
    const width = "16"
    context.req = {
      method: "GET",
      url: `/_next/images/${imagePath ?? ""}?w=${width}`,
      headers: {
        accept:
          "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
      },
      query: { w: width },
      params: { imagePath },
    }

    // When
    const response = await imageOptimizer(context)

    // Then
    expect(response).toMatchSnapshot()
  })

  it("should return a 404 when download the image file fails", async () => {
    // Given
    downloadFileMock.mockReset()
    downloadFileMock.mockRejectedValue("Not found")
    const context = createImageOptimizerMockContext()
    const imagePath = "image.jpeg"
    const width = "16"
    context.req = {
      method: "GET",
      url: `/_next/images/${imagePath ?? ""}?w=${width}`,
      headers: {
        accept:
          "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
      },
      query: { w: width },
      params: { imagePath },
    }

    // When
    const response = await imageOptimizer(context)

    // Then
    expect(response).toEqual({
      status: 404,
      body: "Not Found",
    })
  })
})
