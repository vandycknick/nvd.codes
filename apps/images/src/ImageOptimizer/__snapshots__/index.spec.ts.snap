// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imageOptimizer should return a 200 image jpeg result when no content-type is given 1`] = `
Object {
  "body": Object {
    "data": Array [
      255,
      216,
      255,
      219,
      0,
      67,
      0,
      10,
      7,
      7,
      8,
      7,
      6,
      10,
      8,
      8,
      8,
      11,
      10,
      10,
      11,
      14,
      24,
      16,
      14,
      13,
      13,
      14,
      29,
      21,
      22,
      17,
      24,
      35,
      31,
      37,
      36,
      34,
      31,
      34,
      33,
      38,
      43,
      55,
      47,
      38,
      41,
      52,
      41,
      33,
      34,
      48,
      65,
      49,
      52,
      57,
      59,
      62,
      62,
      62,
      37,
      46,
      68,
      73,
      67,
      60,
      72,
      55,
      61,
      62,
      59,
      255,
      219,
      0,
      67,
      1,
      10,
      11,
      11,
      14,
      13,
      14,
      28,
      16,
      16,
      28,
      59,
      40,
      34,
      40,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      255,
      192,
      0,
      17,
      8,
      0,
      1,
      0,
      1,
      3,
      1,
      34,
      0,
      2,
      17,
      1,
      3,
      17,
      1,
      255,
      196,
      0,
      21,
      0,
      1,
      1,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      4,
      255,
      196,
      0,
      20,
      16,
      1,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      255,
      196,
      0,
      21,
      1,
      1,
      1,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      6,
      7,
      255,
      196,
      0,
      20,
      17,
      1,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      255,
      218,
      0,
      12,
      3,
      1,
      0,
      2,
      17,
      3,
      17,
      0,
      63,
      0,
      144,
      1,
      85,
      9,
      255,
      217,
    ],
    "type": "Buffer",
  },
  "headers": Object {
    "Cache-Control": "public, max-age=3600, s-maxage=31536000",
    "Content-Type": "image/jpeg",
  },
  "status": 200,
}
`;

exports[`imageOptimizer should return a 200 image jpeg result when no content-type is given but the filetype is jpeg 1`] = `
Object {
  "body": Object {
    "data": Array [
      255,
      216,
      255,
      219,
      0,
      67,
      0,
      10,
      7,
      7,
      8,
      7,
      6,
      10,
      8,
      8,
      8,
      11,
      10,
      10,
      11,
      14,
      24,
      16,
      14,
      13,
      13,
      14,
      29,
      21,
      22,
      17,
      24,
      35,
      31,
      37,
      36,
      34,
      31,
      34,
      33,
      38,
      43,
      55,
      47,
      38,
      41,
      52,
      41,
      33,
      34,
      48,
      65,
      49,
      52,
      57,
      59,
      62,
      62,
      62,
      37,
      46,
      68,
      73,
      67,
      60,
      72,
      55,
      61,
      62,
      59,
      255,
      219,
      0,
      67,
      1,
      10,
      11,
      11,
      14,
      13,
      14,
      28,
      16,
      16,
      28,
      59,
      40,
      34,
      40,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      59,
      255,
      192,
      0,
      17,
      8,
      0,
      1,
      0,
      1,
      3,
      1,
      34,
      0,
      2,
      17,
      1,
      3,
      17,
      1,
      255,
      196,
      0,
      21,
      0,
      1,
      1,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      4,
      255,
      196,
      0,
      20,
      16,
      1,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      255,
      196,
      0,
      21,
      1,
      1,
      1,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      6,
      7,
      255,
      196,
      0,
      20,
      17,
      1,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      255,
      218,
      0,
      12,
      3,
      1,
      0,
      2,
      17,
      3,
      17,
      0,
      63,
      0,
      144,
      1,
      85,
      9,
      255,
      217,
    ],
    "type": "Buffer",
  },
  "headers": Object {
    "Cache-Control": "public, max-age=3600, s-maxage=31536000",
    "Content-Type": "image/jpeg",
  },
  "status": 200,
}
`;

exports[`imageOptimizer should return a 200 image png result when no content-type is given but the filetype is png 1`] = `
Object {
  "body": Object {
    "data": Array [
      137,
      80,
      78,
      71,
      13,
      10,
      26,
      10,
      0,
      0,
      0,
      13,
      73,
      72,
      68,
      82,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      1,
      8,
      6,
      0,
      0,
      0,
      31,
      21,
      196,
      137,
      0,
      0,
      0,
      9,
      112,
      72,
      89,
      115,
      0,
      0,
      11,
      19,
      0,
      0,
      11,
      19,
      1,
      0,
      154,
      156,
      24,
      0,
      0,
      0,
      13,
      73,
      68,
      65,
      84,
      8,
      215,
      99,
      248,
      236,
      35,
      241,
      9,
      0,
      5,
      215,
      2,
      74,
      175,
      167,
      19,
      196,
      0,
      0,
      0,
      0,
      73,
      69,
      78,
      68,
      174,
      66,
      96,
      130,
    ],
    "type": "Buffer",
  },
  "headers": Object {
    "Cache-Control": "public, max-age=3600, s-maxage=31536000",
    "Content-Type": "image/png",
  },
  "status": 200,
}
`;

exports[`imageOptimizer should return a 200 image unknown result when no content-type is given but the filetype is unknown 1`] = `
Object {
  "body": Object {
    "data": Array [
      137,
      80,
      78,
      71,
      13,
      10,
      26,
      10,
      0,
      0,
      0,
      13,
      73,
      72,
      68,
      82,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      1,
      8,
      6,
      0,
      0,
      0,
      31,
      21,
      196,
      137,
      0,
      0,
      0,
      9,
      112,
      72,
      89,
      115,
      0,
      0,
      11,
      19,
      0,
      0,
      11,
      19,
      1,
      0,
      154,
      156,
      24,
      0,
      0,
      0,
      13,
      73,
      68,
      65,
      84,
      8,
      215,
      99,
      248,
      236,
      35,
      241,
      9,
      0,
      5,
      215,
      2,
      74,
      175,
      167,
      19,
      196,
      0,
      0,
      0,
      0,
      73,
      69,
      78,
      68,
      174,
      66,
      96,
      130,
    ],
    "type": "Buffer",
  },
  "headers": Object {
    "Cache-Control": "public, max-age=3600, s-maxage=31536000",
    "Content-Type": "image/unknown",
  },
  "status": 200,
}
`;

exports[`imageOptimizer should return a 200 image webp result when no content-type is given but the filetype is webp 1`] = `
Object {
  "body": Object {
    "data": Array [
      82,
      73,
      70,
      70,
      88,
      0,
      0,
      0,
      87,
      69,
      66,
      80,
      86,
      80,
      56,
      88,
      10,
      0,
      0,
      0,
      16,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      65,
      76,
      80,
      72,
      2,
      0,
      0,
      0,
      0,
      242,
      86,
      80,
      56,
      32,
      48,
      0,
      0,
      0,
      240,
      1,
      0,
      157,
      1,
      42,
      1,
      0,
      1,
      0,
      2,
      64,
      56,
      37,
      160,
      2,
      116,
      186,
      1,
      248,
      0,
      4,
      200,
      0,
      0,
      254,
      208,
      202,
      255,
      205,
      129,
      92,
      215,
      62,
      255,
      127,
      251,
      75,
      63,
      74,
      199,
      210,
      177,
      252,
      42,
      0,
      0,
      0,
    ],
    "type": "Buffer",
  },
  "headers": Object {
    "Cache-Control": "public, max-age=3600, s-maxage=31536000",
    "Content-Type": "image/webp",
  },
  "status": 200,
}
`;

exports[`imageOptimizer should return a 200 image webp result when the request contains an accept header 1`] = `
Object {
  "body": Object {
    "data": Array [
      82,
      73,
      70,
      70,
      88,
      0,
      0,
      0,
      87,
      69,
      66,
      80,
      86,
      80,
      56,
      88,
      10,
      0,
      0,
      0,
      16,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      65,
      76,
      80,
      72,
      2,
      0,
      0,
      0,
      0,
      242,
      86,
      80,
      56,
      32,
      48,
      0,
      0,
      0,
      240,
      1,
      0,
      157,
      1,
      42,
      1,
      0,
      1,
      0,
      2,
      64,
      56,
      37,
      160,
      2,
      116,
      186,
      1,
      248,
      0,
      4,
      200,
      0,
      0,
      254,
      208,
      202,
      255,
      205,
      129,
      92,
      215,
      62,
      255,
      127,
      251,
      75,
      63,
      74,
      199,
      210,
      177,
      252,
      42,
      0,
      0,
      0,
    ],
    "type": "Buffer",
  },
  "headers": Object {
    "Cache-Control": "public, max-age=3600, s-maxage=31536000",
    "Content-Type": "image/webp",
  },
  "status": 200,
}
`;
